src/components/CCPForm/CCPFormPage.tsx

```diff
@@ -51,26 +55,28 @@ const useStyles = makeStyles({
 
 const CCPFormPage = ({
   match: {
-    params: { eventID, ccpID },
+    params: { eventId, ccpId },
   },
   mode,
 }: {
   match: {
     params: {
-      eventID: string;
-      ccpID?: string;
+      eventId: string;
+      ccpId?: string;
     };
   };
   mode: string;
 }) => {
   const classes = useStyles();
   const history = useHistory();
 
-  const { data } = useQuery(mode === 'new' ? GET_ALL_CCPS : GET_CCP_BY_ID, {
-    variables: { id: ccpID },
-  });
+  const { data } = useQuery(
+    mode === 'edit' ? GET_CCP_BY_ID : GET_CCPS_BY_EVENT_ID,
+    {
+      variables: { id: ccpId, eventId },
+    }
+  );
```

Comment:
Technically, we don't need to run the `GET_CCPS_BY_EVENT_ID` query, but I added it here for two reasons:
1. We can't conditionally call hooks, so we have to pass something valid to `useQuery` (for both mode new and edit)
2. In `addCCP`, we call `cache.readQuery` to get `collectionPointsByEvent` and add the new CCP to the cache. `readQuery` throws an error if the data isn't available in the cache and never attempts to fetch data from the server (see [here](https://www.apollographql.com/docs/react/caching/cache-interaction/#readquery)). Calling `useQuery` here basically guarantees that the data we need will be in the cache (and the error won't be thrown by `readQuery`).

Reply:
thanks for following up, it's nice that the `readQuery` call is not as expensive as going to the server